framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -618,10 +618,12 @@ protected void checkPostconditions(MethodTree node, ExecutableElement methodElem
                     // check anything
                 } else {
                     CFAbstractValue<?> value = exitStore.getValue(expr);
-                    AnnotationMirror inferredAnno =
-                            value == null
-                                    ? null
-                                    : value.getType().getAnnotationInHierarchy(annotation);
+                    AnnotationMirror inferredAnno = null;
```

Comment:
This code for going from a non-null value to the inferred AnnotationMirror is duplicated three times in this class.
If this is local to the class, put it in a private method. If similar code is outside, should it be a public method here or elsewhere?